WordPress/Classes/Models/ReaderSiteTopic+Lookup.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+
+extension ReaderSiteTopic {
+
+    /// Find a site topic by its site id
+    ///
+    /// - Parameter siteID: The site id of the topic
+    static func lookup(withSiteID siteID: NSNumber, in context: NSManagedObjectContext) throws -> ReaderSiteTopic? {
+        let request = NSFetchRequest<ReaderSiteTopic>(entityName: ReaderSiteTopic.classNameWithoutNamespaces())
+        request.predicate = NSPredicate(format: "siteID = %@", siteID)
+        request.fetchLimit = 1
+        return try context.fetch(request).first
+    }
+
+    /// Find a site topic by its site id
+    ///
+    /// - Parameter siteID: The site id of the topic
+    @objc(lookupWithSiteID:inContext:)
+    static func objc_lookup(withSiteID siteID: NSNumber, in context: NSManagedObjectContext) -> ReaderSiteTopic? {
```

Comment:
Here, we have a duplicated function, one for Swift and one for Objective-C.

In the code above, a different pattern is at use that would result here in a single function with signature

```swift
@objc(lookupWithSiteID:inContext:error:)
static func lookup(withSiteID siteID: NSNumber, in context: NSManagedObjectContext) throws -> ReaderSiteTopic? {
```

Is there any advantage in using a different approach here than in `ReaderAbstractTopic+Lookup`?

Reply:
The compiler doesn't allow it, because the return value is an Optional type. The compile can only create an Objective-C interface for a throwing function which returns a _non-optional_ value. Here is the error generated by the compiler:
> Throwing method cannot be marked @objc because it returns a value of optional type 'ReaderAbstractTopic?'; 'nil' indicates failure to Objective-C